python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -207,17 +207,19 @@ def _get_table_slice(
                     )
 
                 if isinstance(context.asset_partitions_def, MultiPartitionsDefinition):
-                    multi_partition_key_mapping = cast(
-                        MultiPartitionKey, context.asset_partition_key
-                    ).keys_by_dimension
+                    partition_keys = context.asset_partition_keys
                     for part in context.asset_partitions_def.partitions_defs:
-                        partition_key = multi_partition_key_mapping[part.name]
                         if isinstance(part.partitions_def, TimeWindowPartitionsDefinition):
-                            partitions = part.partitions_def.time_window_for_partition_key(
-                                partition_key
-                            )
+                            partitions = context.asset_partitions_time_window
```

Comment:
might need to manually collect the time stamps and then create the range instead of calling this fn